Setup stylesheets

scara82 opened the pull request:
**This branch comes from the ui-interaction branch (open PR) because I needed all the elements for styling**

This PR:
- Creates the files/folders structure for the style with 4 folders in stylesheet folder: base, components, library and patterns
- Add style for:
    - Form to add a new family
    - Table for the list of all the families
    - Introduces a new partial to render just the header of the families list table
    - Style for all the above
    - First style for the header
    - Add some class to elements of table/form for the styling
    - Add translation for families/index

Thanks! :)

Commit:
Add spec files for controller testing

Commit:
Add responders gem

Commit:
Add simple form gem and flash messages to rend it

Commit:
Add families routes for new, create, index and root path with corresponding methods in the controller

Commit:
Add in Domus module repsonders_controller and action_controller config

Commit:
Add specs for families#create

Commit:
Add features spec folder and features/families_spec.rb file

Commit:
Rubocop fixes

Commit:
test

Commit:
Add style for flash messages

Commit:
Add style for the form

Commit:
Add stylesheets/layout folder and first header style, add the _header.html.erb into application.html.erb

Commit:
Create folder system for scss. Create applications.scss, base/components/library/pattern folders and transfer header and form style

Commit:
Add families index translations

Commit:
Add a partial for the table header and create the table for families#index

Commit:
Add style for the families table

Commit:
Deleted unused _lists.scss file

Commit:
Add class to simple_form element for styling

Commit:
Change style for the families_new form

Commit:
Add files for patterns/_titles for titles style

Commit:
Add class title for titles in families/index.html.erb and families/new.html.erb

Commit:
Add _bottons.scss for buttons style

Commit:
Move the button style form _form.scss to his file _button.scss, correct typo for buttons in _patterns.scss

Commit:
Add style for link showing as a button in _buttons.scss

Commit:
Add class link_button adn button to the link for creating a new family in families/index.html.erb

apalski commented on config/locales/simple_form.en.yml:
```diff
@@ -0,0 +1,20 @@
+        bus_driver: "Bus Driver"
+        active: "Active"
+        preferred_gender: "Proferred Gender"
+        submit: "Add"
```
double quotes on yes line 3, no line 4

scara82 replied:
Oh thanks! :)

apalski commented on lib/templates/erb/scaffold/_form.html.erb:
```diff
@@ -0,0 +1,13 @@
+</div>
+
+<div class="form-actions">
+  <%%= f.button :submit %>
```
do you need a translation for the submit button?

scara82 replied:
Yes I did all the translations but it's in the other PR, I was trying to get this the most possible "just style".. ;) But thanks for all the comment! :)

apalski commented on config/locales/responders.en.yml:
```diff
@@ -0,0 +1,12 @@
+         alert: "%{resource_name} could not be updated."
+       destroy:
+         notice: "%{resource_name} was successfully destroyed."
+         alert: "%{resource_name} could not be destroyed."
```
I figure you're going to add a responders.it.yml file down the track?

apalski commented on spec/controllers/families_controller_spec.rb:
```diff
@@ -0,0 +1,28 @@
+        expect(flash[:alert])
+          .to match(
+            I18n.t("flash.actions.create.alert", resource_name: "Family")
+          )
```
You will probably need a translation for your resource_name (I figure I will need to do the same lol)

apalski commented on spec/features/admin_adds_a_family_spec.rb:
```diff
@@ -0,0 +1,16 @@
+
+    expect(page).to have_text(
+      I18n.t("flash.actions.create.notice", resource_name: "Family")
+    )
```
Same here, for the resource_name translation

apalski commented:
Hey Marta, I can't comment much on the style content as I haven't worked on this sort of thing for quite a while. I've added a few comments that I hope will help.

elle commented on .rubocop.yml:
```diff
@@ -12,6 +12,10 @@ Layout/EmptyLineBetweenDefs:
   Enabled: true
+Metrics/MethodLength:
+  Exclude:
+    - "app/controllers/families_controller.rb"
```
This should not be here. Which method is it complaining about? and what is rubicon output?

scara82 replied:
It's complaining about the family_params that is too long and I was not sure how to  fix it. Any suggestion for a better way please?

```
def family_params
    params.require(:family).permit(
      :first_name,
      :last_name,
      :address,
      :animals,
      :capability,
      :preferred_gender,
      :prof,
      :bus_driver,
      :active
    )
  end
```

elle replied:
Maybe change the limit for method length?

elle commented on app/assets/stylesheets/application.scss:
```diff
@@ -0,0 +1,2 @@
+@import "base/_base.scss";
+@import "components/_components.scss";
```
👍

elle commented on app/assets/stylesheets/base/_base.scss:
```diff
@@ -0,0 +1,2 @@
+@import "./_forms.scss";
```
I think you don't to explicitly use `./` here.

scara82 replied:
👌

elle commented on app/assets/stylesheets/base/_forms.scss:
```diff
@@ -0,0 +1,33 @@
+  list-style-type: none;
+  max-width: 800px;
+  margin: 0 auto;
+  background-color: #dce0f2;
```
Alphabetic order for the css definitions?

scara82 replied:
👌

elle commented on app/assets/stylesheets/base/_tables.scss:
```diff
@@ -0,0 +1,48 @@
+.table {
+  display: flex;
+  flex-flow: column nowrap;
+  font-size: .8rem;
```
0.8rem?

scara82 replied:
Ops.. :) Changed into em, thanks!

elle commented on app/assets/stylesheets/base/_tables.scss:
```diff
@@ -0,0 +1,48 @@
+}
+
+.tr:nth-of-type(even) {
+  background-color: #f2f2f2;
```
A suggestion to extract the colours into a colours library and use variables here?
That way, you define a specific shade of grey once, and reuse it everywhere.

scara82 replied:
Done, thanks for the suggestion! 👌

elle commented on app/assets/stylesheets/patterns/_buttons.scss:
```diff
@@ -0,0 +1,16 @@
+  background: #333;
+  color: #f2f2f2;
+  text-transform: uppercase;
+  letter-spacing: .09em;
```
Leading zero?

scara82 replied:
Is it too weird?? I like the letters very close in the buttons :)

elle commented on app/views/families/_families_table_header.html.erb:
```diff
@@ -0,0 +1,13 @@
+<div class="table">
+  <div class="tr th">
+      <div class="td"><%= t ".application.families.show.last_name" %></div>
```
Parens?

elle replied:
Also, usually I use here `Family.human_attribute_name(:last_name)`

Ref: 
- https://stackoverflow.com/questions/4786733/human-attribute-names-in-rails-3-x#7593892
- https://apidock.com/rails/ActiveRecord/Base/human_attribute_name/class

scara82 replied:
Nice to know! I used it, thanks!

elle commented on app/views/families/_family.html.erb:
```diff
@@ -0,0 +1,13 @@
+<div class="table">
+    <div class="tr">
+      <div class="td" style="justify-content: center;">  <%= family.last_name %> </div>
```
Why is the style in the view? can we move it to a stylesheet?

elle replied:
Also double space

scara82 replied:
👌

elle commented on app/views/families/index.html.erb:
```diff
@@ -0,0 +1,4 @@
+<p class="title"><%= t ".title" %></p>
```
parens?

scara82 replied:
👌

elle commented on app/views/layouts/_header.html.erb:
```diff
@@ -0,0 +1,17 @@
+<div class="header">
```
Since this really is a navigation, what do you think about using more semantic markup?
i.e.
```
<nav>
  <ul>
    <li class="header-item><%= link_to "Home page", root_path %>
    ...
  </ul>
</nav>
```

scara82 replied:
👌

elle commented on app/views/layouts/_header.html.erb:
```diff
@@ -0,0 +1,17 @@
+<div class="header">
+  <span class="header-item">
+    Homepage
```
Use translations?

scara82 replied:
Done, thanks!

Commit:
Delete the useless ./ form the path

Commit:
List in alpabetical order

Commit:
Eliminate the ./ from the path for _patterns.scss and _components.scss

Commit:
Change style to use global variables for colour and add _borders.scss and _colors.scss for same reason

Commit:
Move the style into the more appropriate _table.scss file

Commit:
Replace translation with human_attribute_name method

Commit:
Change tag for the nav bar to use nav/ul/li

Commit:
New style for the navigation bar

Commit:
Add parentesis for tranlsation

Commit:
Add translations for nav bar

elle commented on app/assets/stylesheets/_flash.css:
```diff
@@ -0,0 +1,8 @@
+  border: 1px solid red;
+  width: fit-content;
+  margin: 20px auto;
+  font-weight: bold;
```
Alphabetic order?

elle commented on app/assets/stylesheets/base/_tables.scss:
```diff
@@ -0,0 +1,42 @@
+  @include border-bottom-style;
+  display: flex;
+  flex-flow: column nowrap;
+  font-size: .8em;
```
Missing a zero => 0.8em

elle commented on app/assets/stylesheets/base/_tables.scss:
```diff
@@ -0,0 +1,42 @@
+  font-size: .8em;
+  margin: 0.5em;
+  line-height: 1.5;
+  flex: 1 1 auto;
```
Alphabetical order?

elle commented on app/assets/stylesheets/components/_header.scss:
```diff
@@ -0,0 +1,25 @@
+  @include header-style;
+  max-width: 95%;
+  height: 50px;
+  margin: 0 auto;
```
Alphabetical order?

elle commented on app/assets/stylesheets/base/_tables.scss:
```diff
@@ -0,0 +1,42 @@
+@import "patterns/_borders";
```
You don't need to import the individual files in each stylesheet. Just import all of them in `application.scss`

elle commented on app/assets/stylesheets/components/_header.scss:
```diff
@@ -0,0 +1,25 @@
+@import "patterns/_colors";
```
Same here ^

elle commented on app/assets/stylesheets/patterns/_borders.scss:
```diff
@@ -0,0 +1,9 @@
+@mixin border-style {
+  $border: 1px solid #d0d0d0 !global;
```
What is `!global` for?

elle commented on app/assets/stylesheets/patterns/_patterns.scss:
```diff
@@ -0,0 +1,4 @@
+@import "_titles.scss";
```
No need to indicate the `_`

elle commented on app/assets/stylesheets/patterns/_titles.scss:
```diff
@@ -0,0 +1,5 @@
+.title{
```
Space between word and {

Commit:
Remove leftovers from rebase

Commit:
Lock pg version

- to avoid the following error:
Gem::LoadError: can't activate pg (~> 0.18),
already activated pg-1.0.0. Make sure all dependencies are added to Gemfile.

Commit:
Regenrate schema.rb

Commit:
Use rails' link_to helper method

Commit:
Clean rails_helper

Commit:
Clean up stylesheet imports and variable calls

Commit:
Remove unused coffee script file

Commit:
Semantic markup for tables

Commit:
Move header partial to a more approrpiate loc

Commit:
Remove duplication of links

Commit:
Remove duplication of html_wrapper call
